### PR TITLE
Rewrite the legacy token tests without `@solana/spl-token`

### PR DESCRIPTION
**Pull request by steveluscher**

### CODE REVIEW

**Comment by steveluscher** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2821?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2817** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2817?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2816** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2816?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2821** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2821?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2819** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2819?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2815](https://github.com/solana-labs/solana-web3.js/pull/2815) <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2815?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @steveluscher and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/library-legacy/test/connection.test.ts [hunk 1]
```diff
@@ -5175,14 +5147,14 @@ describe('Connection', function () {
...
-        expect(accountsWithMintFilter).to.have.length(2);
+        expect(accountsWithMintFilter).to.have.length(1);
 
         const accountsWithProgramFilter = (
           await connection.getTokenAccountsByOwner(testOwnerKeypair.publicKey, {
             programId: TOKEN_PROGRAM_ID,
           })
         ).value;
-        expect(accountsWithProgramFilter).to.have.length(3);
+        expect(accountsWithProgramFilter).to.have.length(1);
```
Comment (steveluscher) [inline packages/library-legacy/test/connection.test.ts:5157]: This is not _exactly_ the same as the prior test, but it's close enough. We're not meant to be testing the underlying RPC here.

File: packages/library-legacy/test/connection.test.ts [hunk 2]
```diff
@@ -4919,90 +4919,61 @@ describe('Connection', function () {
...
+        '7MbpdfJa5xqwexkp6WUvkYHTPo4VgxYACDBNFWYLwCdo',
+      );
+      const testOwnerKeypair = Keypair.fromSecretKey(
+        new Uint8Array([
+          153, 120, 247, 45, 160, 119, 144, 219, 220, 209, 73, 91, 210, 102, 31,
+          136, 155, 12, 68, 27, 226, 215, 61, 214, 10, 245, 247, 180, 236, 63,
+          100, 202, 140, 247, 112, 54, 120, 32, 168, 118, 72, 115, 190, 34, 171,
+          126, 15, 119, 252, 173, 50, 173, 8, 10, 96, 239, 21, 32, 94, 67, 37,
+          43, 145, 249,
+        ]),
```
Comment (buffalojoec) [inline packages/library-legacy/test/connection.test.ts:4935]: I'm wondering if this should be a fixture somewhere, like `scripts/keypairs`.

File: packages/library-legacy/test/connection.test.ts [hunk 3]
```diff
@@ -4919,90 +4919,61 @@ describe('Connection', function () {
...
+              {
+                pubkey: testOwnerKeypair.publicKey,
+                isSigner: true,
+                isWritable: false,
+              },
+            ],
+            programId: new PublicKey(
+              'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA',
+            ),
+            data: Buffer.from([3, 1, 0, 0, 0, 0, 0, 0, 0]),
```
Comment (buffalojoec) [inline packages/library-legacy/test/connection.test.ts:4968]: Maybe just add a comment here for clarity on the bytes?

File: packages/library-legacy/test/connection.test.ts [hunk 4]
```diff
@@ -4919,90 +4919,61 @@ describe('Connection', function () {
...
       const newAccount = PublicKey.unique();
 
-      let payerKeypair = new Keypair();
-      let testTokenMintPubkey: PublicKey;
-      let testOwnerKeypair: Keypair;
-      let testTokenAccountPubkey: PublicKey;
-      let testSignature: TransactionSignature;
+      const payerPublicKey = new PublicKey(
+        '6bffFjUMnnk3tXFJAS8pp9tz6m7oJAd27vm9rKqnLtVv',
+      );
```
Comment (buffalojoec) [inline packages/library-legacy/test/connection.test.ts:4924]: Is this even needed anymore? Looks like no.

Comment (steveluscher) [reply packages/library-legacy/test/connection.test.ts:4923]: It's used in one dumb place and I was too lazy to see if I could replace it. Let me see.

Comment (steveluscher) [reply packages/library-legacy/test/connection.test.ts]: Gone!

File: packages/library-legacy/test/connection.test.ts (hunk 2 above)
Comment (steveluscher) [reply packages/library-legacy/test/connection.test.ts:4934]: I at least wrote a comment to note what the pubkey is and where to find the fixture.

**Comment by steveluscher** [comment]
### Merge activity

* **Jun 14, 11:10 AM PDT**: @steveluscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2821).

**Pull request merged**